files/common/lib/open-iscsi/unique-name-check.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 by Delphix. All rights reserved.
+#
+
+#
+# This script generates a unique initiator name based on the system-uuid
+#
+
+PATH=/bin:/usr/bin/:/usr/sbin
+
+NAME_FILE="/etc/iscsi/initiatorname.iscsi"
+AUTHORITY="2008-07.com.delphix"
+DMI_KEYWORD="system-uuid"
+
+#
+# IBMcloud changes the system-uuid whenever the engine is
+# power-cycled but preserves the chassis-asset-tag-uuid. For that
+# cloud we use that property to obtain a UUID.
+#
+if [[ "$(dmidecode -s chassis-asset-tag)" == "ibmcloud" ]]; then
+	DMI_KEYWORD="baseboard-asset-tag"
+fi
+
+system_uuid=$(dmidecode -s $DMI_KEYWORD | awk '{print tolower($0)}')
```

Comment:
It feels like this bit of code could live in its own script delivered by delphix-platform. Then, the duplicate code in dlpx-app-gate (from where this was copied) could be removed and replaced with an invocation of this script.

Reply:
good point.  I'll add the script and then file a JIRA to track moving app-gate to switch over.